contracts/admin/src/lib.rs

```diff
@@ -0,0 +1,225 @@
+};
+use versioning::{version, VersionSegment};
+
+use self::{
+    common::{GeneralContractsGroup, SpecializedContractsGroup},
+    error::ContractError,
+    msg::{
+        ContractsMigrateIndividual, ExecuteMsg, GeneralContractsMaybeMigrateIndividual,
+        GeneralContractsMigrateIndividual, InstantiateMsg, MigrateContractsData, MigrateMsg,
+        SpecializedContractsMaybeMigrate, SpecializedContractsMaybeMigrateIndividual,
+        SpecializedContractsMigrateIndividual, SudoMsg,
+    },
+    state::{
+        add_specialized_contracts_group, load_and_remove_migration_release,
+        load_general_contract_addrs, load_specialized_contract_addrs, store_contract_addrs,
+        store_migration_release, SpecializedContractAddrsIter,
+    },
+};
+
+pub mod common;
+pub mod error;
+pub mod msg;
+pub mod state;
+
+// version info for migration info
+// const CONTRACT_STORAGE_VERSION_FROM: VersionSegment = 0;
+const CONTRACT_STORAGE_VERSION: VersionSegment = 0;
+
+#[cfg_attr(feature = "contract-with-bindings", entry_point)]
+pub fn instantiate(
+    deps: DepsMut<'_>,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    versioning::initialize(deps.storage, version!(CONTRACT_STORAGE_VERSION))?;
+
+    platform::contract::validate_addr(&deps.querier, &msg.general_contracts.profit)?;
+    platform::contract::validate_addr(&deps.querier, &msg.general_contracts.timealarms)?;
+    platform::contract::validate_addr(&deps.querier, &msg.general_contracts.treasury)?;
+
+    for group in msg.specialized_contracts.values() {
+        platform::contract::validate_addr(&deps.querier, &group.dispatcher)?;
+        platform::contract::validate_addr(&deps.querier, &group.leaser)?;
+        platform::contract::validate_addr(&deps.querier, &group.lpp)?;
+        platform::contract::validate_addr(&deps.querier, &group.oracle)?;
+    }
+
+    store_contract_addrs(
+        deps.storage,
+        msg.general_contracts,
+        msg.specialized_contracts,
+    )?;
+
+    Ok(Response::default())
+}
+
+#[cfg_attr(feature = "contract-with-bindings", entry_point)]
+pub fn migrate(deps: DepsMut<'_>, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
+    versioning::update_software(deps.storage, version!(CONTRACT_STORAGE_VERSION))?;
+
+    Ok(Response::default().set_data(to_binary(sdk::RELEASE_VERSION)?))
+}
+
+#[cfg_attr(feature = "contract-with-bindings", entry_point)]
+pub fn execute(
+    deps: DepsMut<'_>,
+    env: Env,
+    _info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        #[cfg(any(debug_assertions, test, feature = "admin_contract_exec"))]
+        ExecuteMsg::LocalNetSudo { sudo: msg } => sudo(deps, env, msg),
+    }
+}
+
+#[cfg_attr(feature = "contract-with-bindings", entry_point)]
+pub fn sudo(deps: DepsMut<'_>, env: Env, msg: SudoMsg) -> Result<Response, ContractError> {
+    match msg {
+        SudoMsg::RegisterSymbolGroup {
```

Comment:
do not get the aim of this message, could you elaborate? discuss it offline is also an option

Reply:
RegisterSymbolGroup -> RegisterLpnContracts